Lib/TestRunnerBase.py

```diff
@@ -0,0 +1,338 @@
+from __future__ import print_function
+
+import glob
+import sys
+import os
+import multiprocessing
+import subprocess
+import image_compare
+import codecs
+import time
+import webbrowser
+import shlex
+
+from Util import *
+from Const import *
+
+class TestRunnerBase:
+
+    def __init__(self, test_suite_name, args, get_sample_data=False):
+        self.test_suite_name = test_suite_name
+        self.run_options = 0x00000000
+        if args.html:
+            self.run_options |= OPT_GENERATE_HTML
+        if args.package:
+            self.run_options |= OPT_PACKAGE_RESULT
+        if args.git:
+            self.run_options |= OPT_GET_BASELINE
+        if args.failed_only:
+            self.run_options |= OPT_FAILED_ONLY
+        if args.no_vtk_ui:
+            self.run_options |= OPT_NO_VTK_UI
+        if args.verbosity:
+            self.verbosity = args.verbosity
+        else:
+            self.verbosity = 1
+        if args.vtk:
+            self.vtk_channels = args.vtk
+        if args.cpus:
+            self.ncpus = args.cpus
+        else:
+            self.ncpus = 1
+
+        if args.attributes:
+            self.nosetests_attr = args.attributes
+        else:
+            self.nosetests_attr = []
+
+        if get_sample_data == True:
+            download_sample_data_files(os.path.join(sys.prefix, "share", 
```

Comment:
let's have the file name sent as part of base class initialization

Reply:
Done. 
Made the  test_data_files_info to be an argument of the TestRunnerBase's __init()